src/lib/caches/Items.js

```diff
@@ -75,41 +75,90 @@ module.exports = class ItemsCache {
       hydrate();
       base = ItemsCache.#cache.getKey(`${language}-${key}`);
     }
+    // Allow nested keys separated by periods
+    const bys = by.split('.');
     let filtered = base;
     if (term && max !== 1) {
       filtered = base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
         .map((item) => {
-          if (!(item && item[by])) return undefined;
-          if (item[by].toLowerCase().includes(term.toLowerCase())) {
+          const values = this.#parseNestedKey(item, bys);
+          if (!(item && values)) return undefined;
+          // parseNestedKey returns an array, so check all values in that array
+          if (values.find((entry) => entry.toString().toLowerCase().includes(term.toString().toLowerCase()))) {
             return item;
           }
           return undefined;
         })
         .filter((a) => a)
-        .flat();
+        .flat(Infinity);
     }
     if (term && max === 1) {
       let keyDistance;
       let exact = false;
       filtered = undefined;
       base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
         .forEach((item) => {
-          if (item[by].toLowerCase() === term.toLowerCase()) {
+          const values = this.#parseNestedKey(item, bys);
+          // parseNestedKey returns an array, so check all values in that array
+          if (values.find((entry) => entry.toString().toLowerCase() === term.toString().toLowerCase())) {
             filtered = item;
             exact = true;
           }
-
-          if (item[by].toLowerCase().includes(term.toLowerCase()) && !exact) {
-            const distance = item[by].toLowerCase().replace(term.toLowerCase(), '').length;
-            if (!keyDistance || distance < keyDistance) {
-              keyDistance = distance;
-              filtered = item;
-            }
+          if (!exact) {
+            // parseNestedKey returns an array, so check all values in that array
+            values.forEach((entry) => {
+              if (entry.toString().toLowerCase().includes(term.toString().toLowerCase())) {
+                const distance = entry.toString().toLowerCase().replace(term.toString().toLowerCase(), '').length;
+                if (!keyDistance || distance < keyDistance) {
+                  keyDistance = distance;
+                  filtered = item;
+                }
+              }
+            });
           }
         });
     }
     return this.#cleanup(filtered, { remove, only });
   }
+
+  /**
+   * Utility function for following nested keys. Since some keys have values that are
+   * an array of non-keyed objects, this function iterates over those arrays and can
+   * return multiple values for a single key path.
+   * @param {Array} item data to search
```

Comment:
should probably type this on <Item> since we have the types coming in from warframe-items (theoretically)

Reply:
Updated in 12ea7a2